File: go/1669. Merge In Between Linked Lists.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
	beforeA := getNode(list1, a-1)
	afterB := getNode(list1, b+1)
	beforeA.Next = list2
	for beforeA.Next != nil {
		beforeA = beforeA.Next
	}
	beforeA.Next = afterB
	return list1
 }

func getNode(head *ListNode, index int) *ListNode {
	for index > 0 {
		head = head.Next
		index -= 1
	}
	return head
}

File: java/1669. Merge In Between Linked Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode nodeBeforeA = getNode(list1, a-1);
        ListNode nodeAfterB = getNode(list1, b+1);
        nodeBeforeA.next = list2;
        while (nodeBeforeA.next != null) nodeBeforeA = nodeBeforeA.next;
        nodeBeforeA.next = nodeAfterB;
        return list1;
    }
    private ListNode getNode(ListNode head, int count) {
        while (count-- > 0) head = head.next;
        return head;
    }
}